src/averageAges.js

```diff
@@ -36,8 +45,22 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = !!withChildren
+    ? people.filter(person =>
+      person.sex === 'f' && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === FEMALE);
+
+  const womenTotalAge = women.reduce((acc, woman) => {
+    return acc + (woman.died - woman.born);
+  }, 0);
+
+  womenTotalAge.sort().reverse();
```

Comment:
The operation `womenTotalAge.sort().reverse();` is not applicable here since `womenTotalAge` is a number, not an array. This line should be removed as it does not contribute to the function's logic.